specification/keyvault/data-plane/TypeSpec/routes.tsp

```diff
@@ -0,0 +1,278 @@
+ * operation requires the secrets/set permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Sets a secret in a specified key vault.")
+@route("/secrets/{secretName}")
+@put
+op setSecret is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The name of the secret. The value you provide may be copied globally for the
+     * purpose of running the service. The value provided should not include
+     * personally identifiable or sensitive information.
+     */
+    @pattern("^[0-9a-zA-Z-]+$")
+    @path
+    secretName: string;
+
+    /**
+     * The parameters for setting the secret.
+     */
+    #suppress "deprecated" "Property flattening is supported for legacy scenarios like Key Vault's"
+    @body
+    @flattenProperty
+    parameters: SecretSetParameters;
+  },
+  SecretBundle
+>;
+
+/**
+ * The DELETE operation applies to any secret stored in Azure Key Vault. DELETE
+ * cannot be applied to an individual version of a secret. This operation requires
+ * the secrets/delete permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Deletes a secret from a specified key vault.")
+@route("/secrets/{secretName}")
+@delete
+op deleteSecret is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The name of the secret.
+     */
+    @path
+    secretName: string;
+  },
+  DeletedSecretBundle
+>;
+
+/**
+ * The UPDATE operation changes specified attributes of an existing stored secret.
+ * Attributes that are not specified in the request are left unchanged. The value
+ * of a secret itself cannot be changed. This operation requires the secrets/set
+ * permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Updates the attributes associated with a specified secret in a given key vault.")
+@route("/secrets/{secretName}/{secretVersion}")
+@patch
+op updateSecret is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The name of the secret.
+     */
+    @path
+    secretName: string;
+
+    /**
+     * The version of the secret.
+     */
+    @path
+    secretVersion: string;
+
+    /**
+     * The parameters for update secret operation.
+     */
+    #suppress "deprecated" "Property flattening is supported for legacy scenarios like Key Vault's"
+    @body
+    @flattenProperty
+    parameters: SecretUpdateParameters;
+  },
+  SecretBundle
+>;
+
+/**
+ * The GET operation is applicable to any secret stored in Azure Key Vault. This
+ * operation requires the secrets/get permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Get a specified secret from a given key vault.")
+@route("/secrets/{secretName}/{secretVersion}")
+@get
+op getSecret is Azure.Core.Foundations.Operation<
```

Comment:
This will create problems for SDKs but I'm not sure how to solve it. We went with making "version" an optional parameter across languages, which means we basically collapsed the routes into one. @lmazuel or @johanste any suggestions? Can client TSPs "fix" this, or is this a case where we're going to generate an incompatible SDK and have to either wrap it or ship a new major version?

Reply:
/cc @pallavit